packages/plugins/rum/src/validate.ts

```diff
@@ -0,0 +1,103 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019-Present Datadog, Inc.
+
+import type { Logger } from '@dd/core/types';
+import chalk from 'chalk';
+
+import { CONFIG_KEY, PLUGIN_NAME } from './constants';
+import type {
+    OptionsWithRum,
+    RumOptions,
+    RumOptionsWithDefaults,
+    SDKOptionsWithDefaults,
+} from './types';
+
+export const validateOptions = (
+    options: Partial<OptionsWithRum>,
+    log: Logger,
+): RumOptionsWithDefaults => {
+    const errors: string[] = [];
+
+    // Validate and add defaults sub-options.
+    const sdkResults = validateSDKOptions(options);
+
+    errors.push(...sdkResults.errors);
+
+    // Throw if there are any errors.
+    if (errors.length) {
+        log.error(`\n  - ${errors.join('\n  - ')}`);
+        throw new Error(`Invalid configuration for ${PLUGIN_NAME}.`);
+    }
+
+    // Build the final configuration.
+    const toReturn: RumOptionsWithDefaults = {
+        ...options[CONFIG_KEY],
+        sdk: undefined,
+    };
+
+    // Fill in the defaults.
+    if (sdkResults.config) {
+        toReturn.sdk = sdkResults.config;
+    }
+
+    return toReturn;
+};
+
+type ToReturn<T> = {
+    errors: string[];
+    config?: T;
+};
+
+export const validateSDKOptions = (
+    options: Partial<OptionsWithRum>,
+): ToReturn<SDKOptionsWithDefaults> => {
+    const red = chalk.bold.red;
+    const validatedOptions: RumOptions = options[CONFIG_KEY] || {};
+    const toReturn: ToReturn<SDKOptionsWithDefaults> = {
+        errors: [],
+    };
+
+    if (validatedOptions.sdk) {
+        // Validate the configuration.
+        if (!validatedOptions.sdk.applicationId) {
+            toReturn.errors.push(`Missing ${red('applicationId')} in the SDK configuration.`);
+        }
+
+        // Check if we have all we need to fetch the client token if necessary.
+        if ((!options.auth?.apiKey || !options.auth?.appKey) && !validatedOptions.sdk.clientToken) {
+            toReturn.errors.push(
+                `Missing ${red('"auth.apiKey"')} and/or ${red('"auth.appKey"')} to fetch missing client token.`,
+            );
+        }
+
+        const sdkWithDefault: SDKOptionsWithDefaults = {
+            applicationId: 'unknown_application_id',
+            allowUntrustedEvents: false,
+            compressIntakeRequests: false,
+            defaultPrivacyLevel: 'mask',
+            enablePrivacyForActionName: false,
+            sessionReplaySampleRate: 0,
+            sessionSampleRate: 100,
+            silentMultipleInit: false,
+            site: 'datadoghq.com',
+            startSessionReplayRecordingManually: false,
+            storeContextsAcrossPages: false,
+            telemetrySampleRate: 20,
+            traceSampleRate: 100,
+            trackingConsent: 'granted',
+            trackLongTasks: false,
+            trackResources: false,
+            trackUserInteractions: false,
+            trackViewsManually: false,
+        };
```

Comment:
Same here: Idk if we should re-code the SDK defaults. Is it possible to get them from the SDK package itself? (I'd prefer to avoid duplicating code, specially if they decide to change their defaults)

Reply:
Fair, I'll look into it.

Reply:
Edit: as you're fully pinning a version of the browser SDK, hard coding values is not needed (as you can decide to update those when updating the underlying RUM version)

Idk if you want to manually check that the default values are the same, so it could still be good, but it's not required